plugin/trino-hive/src/main/java/io/trino/plugin/hive/ion/IonFileWriter.java

```diff
@@ -51,6 +52,7 @@ public IonFileWriter(
             Closeable rollbackAction,
             TypeManager typeManager,
             Optional<CompressionKind> compressionKind,
+            Boolean encoding,
```

Comment:
nit: variable name is ambiguous as a boolean.

I don't know about the trino code style in regards to this, but in general it's better to have a clear type for such flags as a method parameter. e.g. an `IonEncoding` enum, you could even put the logic of what writer to create as a method of the enum. That may be overengineering it, but at the least the name needs to change.